### PR TITLE
refactor: move manifest validation to common package

### PR DESCRIPTION
**Pull request by ZeRego**
Related to: #6209

### Description:

Changes:
- move all logic on dbt manifest validation to common package
- remove duplicate code and files from CLI and Backend packages
 
Kept the scope of this refactor small even though there is more code that should be moved to common.

### CODE REVIEW

File: packages/common/src/dbt/validation.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    `Missing dbt manifest version "${manifestVersion}" in validation.`,
+                ),
+            );
+    }
+
+    const ajv = new Ajv({
+        schemas: [
+            getDbtManifestWithLightdashMetadata(dbtManifestSchema.$id),
+            lightdashMetadataSchema,
+            dbtManifestSchema,
```
Comment (ZeRego) [inline packages/common/src/dbt/validation.ts:96]: Our validation now depends on 3 schemas:
- **dbtManifestSchema**: is the original schema from dbt. The schema file depends on what version of dbt it was used to generate the manifest.
- **lightdashMetadataSchema**: contains the definitions for the Lightdash meta properties
- **getDbtManifestWithLightdashMetadata()**: combines the original dbt schema with Lightdash definitions

File: packages/common/src/dbt/validation.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import { ParseError, UnexpectedServerError } from '../types/errors';
+import assertUnreachable from '../utils/assertUnreachable';
+import lightdashMetadataSchema from './schemas/lightdashMetadata.json';
+import dbtManifestSchemaV8 from './schemas/manifestV8.json';
+import dbtManifestSchemaV9 from './schemas/manifestV9.json';
+
+const LightdashMetadataSchemaRef =
+    'https://schemas.lightdash.com/dbt/manifest.json';
+
+const getDbtManifestWithLightdashMetadata = (dbtSchemaRef: string) => ({
```
Comment (rephus) [inline packages/common/src/dbt/validation.ts:19]: this is ok for  v8 and v9 have the same definitions, but adding support (again) for v7 , or a potentially a different future version, might not look great. What if we create a separate "static" version instead ?

**Review by rephus — COMMENTED** [review]
Good refactoring, this is a lot better. I just have a question about adding v7 again

File: packages/common/src/dbt/validation.ts (hunk 2 above)
Comment (ZeRego) [reply packages/common/src/dbt/validation.ts:19]: yes, that makes sense 👍 
Will change it.

**Pull request merged**